mesh_map/src/mesh_map.cpp

```diff
@@ -1042,41 +1042,33 @@ lvr2::OptionalFaceHandle MeshMap::getContainingFace(Vector& position, const floa
 
 boost::optional<std::tuple<lvr2::FaceHandle, std::array<mesh_map::Vector , 3>,
     std::array<float, 3>>> MeshMap::searchContainingFace(
-    Vector& position, const float& max_dist)
+    Vector& query_point, const float& max_dist)
 {
-  if(auto vH_opt = getNearestVertexHandle(position))
+  if(auto vH_opt = getNearestVertexHandle(query_point))
   {
     auto vH = vH_opt.unwrap();
-    float min_triangle_position_distance = std::numeric_limits<float>::max();
-    std::array<Vector, 3> vertices;
-    std::array<float, 3> bary_coords;
-    lvr2::OptionalFaceHandle opt_fH;
-    for(auto fH : mesh_ptr->getFacesOfVertex(vH))
+    float lowest_distance_found = std::numeric_limits<float>::max();
+    std::array<Vector, 3> closest_face_vertices;
+    std::array<float, 3> bary_coords_on_closest_face;
+    lvr2::OptionalFaceHandle opt_closest_face_handle;
+    for(auto current_face_handle : mesh_ptr->getFacesOfVertex(vH))
     {
-      const auto& tmp_vertices = mesh_ptr->getVertexPositionsOfFace(fH);
-      float dist = 0;
-      std::array<float, 3> tmp_bary_coords;
-      if (mesh_map::projectedBarycentricCoords(position, vertices, tmp_bary_coords, dist)
-          && std::fabs(dist) < max_dist)
-      {
-        return std::make_tuple(fH, tmp_vertices, tmp_bary_coords);
-      }
+      const auto& current_vertices = mesh_ptr->getVertexPositionsOfFace(current_face_handle);
+      float distance_to_current_face = 0;
+      std::array<float, 3> current_bary_coords;
+      const bool is_query_point_in_current_face = mesh_map::projectedBarycentricCoords(query_point, current_vertices, current_bary_coords, distance_to_current_face);
 
-      float triangle_dist = 0;
```

Comment:
After fixing the points above, I got invalid queries like this:
![wrong_goal_pose_mid_fix](https://github.com/naturerobots/mesh_navigation/assets/727012/7c744ad0-5575-4ab3-88f0-2b5b52f5c970)

I am not sure what the `triangle_dist` should be, but I don't see how this is a meaningful distance. So, I decided to use the distance given by the `projectedBarycentricCoords` functions and ended up with the behavior I'd expect.